.travis.yml

```diff
@@ -90,40 +106,49 @@ matrix:
+      curl https://gstreamer.freedesktop.org/data/pkg/osx/1.0.10/gstreamer-1.0-1.0.10-universal.pkg -vlo gstreamer-1.0-1.0.10-universal.pkg;
+      curl https://gstreamer.freedesktop.org/data/pkg/osx/1.0.10/gstreamer-1.0-devel-1.0.10-universal.pkg -vlo gstreamer-1.0-devel-1.0.10-universal.pkg;
     fi
 
 install:
   - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
-    brew tap PX4/px4;
-    brew install px4-dev pkg-config;
-    installer -allowUntrusted -verboseR -pkg /tmp/gstreamer-1.0-1.0.10-universal.pkg -target /;
-    installer -allowUntrusted -verboseR -pkg /tmp/gstreamer-1.0-devel-1.0.10-universal.pkg -target /;
-    export PKG_CONFIG_PATH="/Frameworks/GStreamer.framework/Versions/Current/lib/pkgconfig${PKG_CONFIG_PATH:+:$PKG_CONFIG_PATH}";
-    brew tap osrf/simulation;
-    brew install opencv ${GAZEBO_VERSION};
-    brew cask reinstall xquartz java;
-    sudo -H pip2 install --upgrade pip setuptools;
-    sudo -H pip2 install rospkg pyserial empy toml numpy pandas jinja2;
-    sudo -H pip2 install --upgrade --force-reinstall numpy;
+      pushd googletest;
+      mkdir build;
+      pushd build;
+      cmake ..;
+      make;
+      make install;
+      popd;
+      popd;
+      brew tap PX4/px4;
+      brew install px4-dev;
+      brew tap osrf/simulation;
+      brew install opencv ${GAZEBO_VERSION};
+      sudo -H pip2 install --upgrade pip setuptools;
+      sudo -H pip2 install rospkg pyserial empy toml numpy pandas jinja2;
+      sudo installer -allowUntrusted -verboseR -pkg gstreamer-1.0-1.0.10-universal.pkg -target /;
+      sudo installer -allowUntrusted -verboseR -pkg gstreamer-1.0-devel-1.0.10-universal.pkg -target /;
+    fi
+
+after_install:
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
+      brew cleanup;
+      rm -rf /usr/local/include/mavlink/v2.0/.git googletest/.git;
+      rm gstreamer-1.0-1.0.10-universal.pkg gstreamer-1.0-devel-1.0.10-universal.pkg;
+    fi
+
+before_script:
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
+      export PKG_CONFIG_PATH="/Frameworks/GStreamer.framework/Versions/Current/lib/pkgconfig${PKG_CONFIG_PATH:+:$PKG_CONFIG_PATH}";
     fi
 
 script:
   - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
-    mkdir Build;
-    cd Build && cmake -DENABLE_UNIT_TESTS=On .. && make -j$(sysctl -n hw.physicalcpu) -l$(sysctl -n hw.physicalcpu); make test;
-    else docker run -it --rm -w ${TRAVIS_BUILD_DIR} --env=CCACHE_DIR="${CCACHE_DIR}" --volume=${CCACHE_DIR}:${CCACHE_DIR}:rw --volume=${TRAVIS_BUILD_DIR}:${TRAVIS_BUILD_DIR}:rw ${PX4_DOCKER_REPO} /bin/bash -e -c "${BUILD}";
+      mkdir Build;
+      cd Build && cmake -DENABLE_UNIT_TESTS=On .. && make -j$(sysctl -n hw.physicalcpu) -l$(sysctl -n hw.physicalcpu); make test;
+    else
+      docker run -it --rm -w ${TRAVIS_BUILD_DIR} --env=CCACHE_DIR="${CCACHE_DIR}" --volume=${CCACHE_DIR}:${CCACHE_DIR}:rw --volume=${TRAVIS_BUILD_DIR}:${TRAVIS_BUILD_DIR}:rw ${PX4_DOCKER_REPO} /bin/bash -e -c "${BUILD}";
```

Comment:
@TSC21 what do we use docker for here?

Reply:
To build on the Linux distros.

Reply:
But that's inside the `if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then`?

Reply:
Oh, there is the else :see_no_evil: